### PR TITLE
chore(sdk-lib-mpc): proxy rangeProof (sdk-core) methods to EcdsaRangeProof (sdk-lib-mpc)

### PR DESCRIPTION
**Pull request by zahin-mohammad**
## Description

One pr of many of migrating functions to sdk-lib-mpc.

## Issue Number

BG-77573

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts
Comment (islamaminBitGo) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:1]: rangeProof is still being used to call generateNtilde, can we replace that with EcdsaRangeProof and remove the ./rangeProof import from this file?

Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts:1]: Good catch.

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (rlutzbitgo) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:43]: I still think this comment is misleading at best (if it was copied into the new location for the generateModulus function). If we want to throw instead of just auto-retrying, I think that's fine, but you can't guarantee that multiplying two 1536 bit numbers together will result in a 3072 bit number. 

1111 * 1000 = 1111000 (7 bits)
1111 * 1111 = 11100001 (8 bits)

**Review by rlutzbitgo — DISMISSED** [review]
Looks good to me. I restarted your node 16 tests because one of them timed out (probably worth considering bumping some of these timeouts given how the variance is with things like safe prime generation)

**Comment by rlutzbitgo** [comment]
Given that it failed twice in a row, should probably just go ahead and bump the timeouts for sdk-coin-atom:
```
1) ATOM
@bitgo/sdk-coin-atom:        Recover transactions: success path
@bitgo/sdk-coin-atom:          should recover funds for non-bitgo recoveries:
@bitgo/sdk-coin-atom:      Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

**Review by islamaminBitGo — DISMISSED** [review]

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (islamaminBitGo) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:43]: You bring up a good point, I think we should retry if it doesnt hit 3072 instead of throwing. Our maximum needed bits will always be bitlengthP + bitlengthQ so we can keep retrying until the modulus hits that.

**Comment by zahin-mohammad** [comment]
> > should recover funds for non-bitgo recoveries:
> 
> 🙄 I think someone forgot to mock challenge generation, let me TAL

That person turned out to be me. Now that things are in the new library, the stubs need to be updated.

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:43]: Agreed, but lets make the retry count an optional input. We can add a default value if wed like.

**Review by islamaminBitGo — DISMISSED** [review]

**Comment by zahin-mohammad** [comment]
Please merge once this gets all the needed approvals.

File: modules/sdk-core/src/account-lib/mpc/curves/secp256k1.ts
Comment (sachushaji) [inline modules/sdk-core/src/account-lib/mpc/curves/secp256k1.ts:10]: Why can't we also export the Secp256k1Curve from sdk-lib-mpc along side the deprecation notice?, since making changes in the Secp256k1Curve should also be reflected here until we remove it from here

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts
Comment (sachushaji) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:34]: Are we not adding a generateModulus function exported from sdk-lib-mpc here?

File: modules/sdk-core/src/account-lib/mpc/util.ts
Comment (sachushaji) [inline modules/sdk-core/src/account-lib/mpc/util.ts:38]: Same, lets use it from sdk-lib-mpc and export here along side the deprecation notice so that any changes made to this function in the sdk-lib-mpc would also be reflected here

File: modules/sdk-lib-mpc/src/tss/index.ts
Comment (sachushaji) [inline modules/sdk-lib-mpc/src/tss/index.ts:1]: Are we only going to use this library for ecdsa, Named export can help to distinguish between similar types/ functions between Ecdsa and Eddsa, without it can be confusing

Comment (sachushaji) [reply modules/sdk-lib-mpc/src/tss/index.ts:1]: We can also squash the commits

File: modules/sdk-core/src/account-lib/mpc/curves/secp256k1.ts
Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/curves/secp256k1.ts:10]: Good idea, didn't think about re-exporting from the deprecated file.

Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/curves/secp256k1.ts]: Fixed.

File: modules/sdk-core/src/account-lib/mpc/util.ts
Comment (zahin-mohammad) [reply modules/sdk-core/src/account-lib/mpc/util.ts]: Fixed.

**Review by islamaminBitGo — APPROVED** [review]

**Review by islamaminBitGo — DISMISSED** [review]

**Review by islamaminBitGo — APPROVED** [review]

**Review by sachushaji — APPROVED** [review]
LGTM, Thanks @zahin-mohammad

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**
